python/cuml/svm/linear.pyx

```diff
@@ -452,6 +460,35 @@ cdef class LinearSVMWrapper:
 
         return y
 
+    def decision_function(self, X: CumlArray) -> CumlArray:
+        n_classes = self.classes_.shape[0]
+        # special handling of binary case
+        shape = (X.shape[0],) if n_classes <= 2 else (X.shape[0], n_classes)
+        y = CumlArray.empty(
+            shape=shape,
+            dtype=self.dtype, order='C')
+
+        if self.dtype == np.float32:
+            LinearSVMModel[float].decisionFunction(
+                deref(self.handle),
+                self.params,
+                self.model.float32,
+                <const float*><uintptr_t>X.ptr,
+                X.shape[0], X.shape[1],
+                <float*><uintptr_t>y.ptr)
+        elif self.dtype == np.float64:
+            LinearSVMModel[double].decisionFunction(
+                deref(self.handle),
+                self.params,
+                self.model.float64,
+                <const double*><uintptr_t>X.ptr,
+                X.shape[0], X.shape[1],
+                <double*><uintptr_t>y.ptr)
+        else:
+            raise TypeError('Input data type should be float32 or float64')
```

Comment:
Since this is not only a recommendation, but a requirement:
```suggestion
            raise TypeError('Input data type must be float32 or float64')
```